### PR TITLE
CSP-329-web-not-retrieving-standards

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/SFA.DAS.Roatp.Application/Providers/Queries/GetProviders/ProviderAddressModel.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                AddressLine4 = source.AddressLine4,
+                Town = source.Town,
+                Postcode = source.Postcode,
+                Latitude = source.Latitude,
+                Longitude = source.Longitude,
+                AddressUpdateDate = source.AddressUpdateDate,
+                CoordinatesUpdateDate = source.CoordinatesUpdateDate,
+            };
+        }
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Application/Providers/Queries/GetProviders/ProviderAddressModel.cs:42]: remove spaces

File: src/SFA.DAS.Roatp.Data/Repositories/ProvidersReadRepository.cs [hunk 2]
```diff
@@ -11,23 +13,41 @@ namespace SFA.DAS.Roatp.Data.Repositories
...
-                        .AsNoTracking().SingleOrDefaultAsync(p => p.Ukprn == ukprn);
+            try
+            {
+                return await _roatpDataContext.Providers
+                            .Include(p => p.ProviderAddress)
+                            .AsNoTracking().SingleOrDefaultAsync(p => p.Ukprn == ukprn);
+            }
+            catch(Exception e)
+            {
+                _logger.LogError("Error occured while getting provider {ukprn} error message : {error}", ukprn, e.Message);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Data/Repositories/ProvidersReadRepository.cs:33]: I'd change this to _logger.LogError(e,$"Error occured while getting provider {ukprn}")

It makes all parts of the error available in the log

File: src/SFA.DAS.Roatp.Data/Repositories/ProvidersReadRepository.cs [hunk 3]
```diff
@@ -11,23 +13,41 @@ namespace SFA.DAS.Roatp.Data.Repositories
...
-                        .Include(p => p.Address)
+            try
+            {
+                return await _roatpDataContext.Providers
+                        .Include(p => p.ProviderAddress)
                         .AsNoTracking().ToListAsync();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError("Error occured while getting all providers error message : {error}", e.Message);
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.Data/Repositories/ProvidersReadRepository.cs:48]: see previous suggestion about logging errors

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.Roatp.Data/Repositories/ProvidersWriteRepository.cs [hunk 4]
```diff
@@ -20,5 +22,12 @@ public async Task Create(Provider provider)
             _roatpDataContext.Providers.Add(provider);
             await _roatpDataContext.SaveChangesAsync();
         }
+
+        public async Task<List<Provider>> GetAllProviders()
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Data/Repositories/ProvidersWriteRepository.cs:26]: I don't think this is required as we are mapping ProviderId before saving the address. Remove this method from write repository and use read repository instead.

**Review by DevenKShah — CHANGES_REQUESTED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**